### ISSUE TITLE
WASM module signature verification

### ISSUE THREAD

**Issue by asraa** [istio/istio#32002]
**Describe the feature request**
Include container signatures and verification as part of Istio WASM adoption. The WASM extension will soon be supporting fetching OCI images, this proposal is to add support for specifying a public key to verify the image in the WasmExtension Config (https://github.com/istio/api/pull/1940) and to support verification on download.

Concretely, I'm proposing
1. Support the option to specify verification in the Istio Wasm Extension API’s image settings.
2. Implement signature verification using cosign (https://github.com/sigstore/cosign) in Istio Agent’s Wasm fetching

Cosign can already be used to upload signed OCI images to WebAssemblyHub  etc. It supports generating ECDSA P-256 keys, multiple signatures, and additional annotations to the signed payload. RE other algs, it can support uploading and fetching any type of signatures to be verified with other tools. 

** Additional Stuff **
* Keyless modes: Cosign can also support keyless modes (https://github.com/sigstore/cosign/blob/main/KEYLESS.md) where certs are signed by a root CA (https://github.com/sigstore/fulcio)  and signatures are stored in a transparency log. Verification would involve calling to check the transparency log. I'm not sure if that's off interest, but support can be added. There's also WIP work to support bundling (https://github.com/sigstore/cosign/issues/181) to avoid the network call.
* KMS support: It also provides signing and verifying from a KMS path (GCP KMS support right now)

**Describe alternatives you've considered**
Module signatures verified at load: Signatures can also be directly on the module, I think this should also exist and serves different uses (verify pre-compiled headers, support verification on downloads via HTTP/HTTPS).

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x ] Extensions and Telemetry
[x ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Additional Context**

cc @PiotrSikora @dgn @mathetake

**Comment by zirain** [comment]
not stale

**Comment by jcchavezs** [comment]
I can take this

**Comment by lukehinds** [comment]
did you work on this @jcchavezs ?

**Comment by jcchavezs** [comment]
Yeah I started some work on this. Plan to finish in the week after (some deadlines in the middle)
